### DISCUSSION TITLE
I am a newbee in C#. What about database in c#?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [dotnet/runtime#100963] [Q&A]
I am developing a desktop software that runs on both Windows and Mac, and I need a database to store project information. Previously, I used PHP and MySQL databases. Is there any C# database that I can quickly learn? Also, I use AOT, and it seems that many databases do not support AOT well. Any recommendations?

**Comment by Clockwork-Muse** [comment]
> Previously, I used PHP and MySQL databases. Is there any C# database that I can quickly learn?

Strictly speaking it's the connection to the database and the data framework around it you have to learn.
In C# the default data framework is EFCore, which [you can use with MySQL](https://dev.mysql.com/doc/connector-net/en/connector-net-entityframework-core-example.html) (and a lot of other databases too, like Microsoft's SQL Server or PostgreSQL).

> Also, I use AOT, and it seems that many databases do not support AOT well.

... in this context, I'm reading "AOT" as "ahead of time (compilation)".  The database is unimportant for this (unless maybe  it's something small and in-process, like SQLite).  The data framework may have issues - [EF Core doesn't yet support it](https://github.com/dotnet/efcore/issues/29754), but other frameworks might.

That said, for most exploratory/personal projects, AOT isn't something you should need to worry about.

**Reply by neon-sunset** [reply]
Agree with the comment, just wanted to note there is https://aot.dapperlib.dev now for this kind of deployment.
